R/prepare_stop_level_data.R

```diff
@@ -26,20 +29,14 @@ granulate_to_stop <- function(mbbs) {
   beep()
 }
 
-#' Processes the stop level information left in the species_comments column of mbbs rows imported from ebird
-#' Returns df of mbbs data that had an informative species_comments broken down to the stop level
-#'@importFrom dplyr %>% filter mutate relocate select
-#'@importFrom stringr str_replace_all str_starts str_split str_extract 
-#'@importFrom tidyr pivot_longer
-#'@param mbbs Any mbbs dataset, either the whole survey area or one county
-process_species_comments <- function(mbbs) {
-  
-  #pull rows that have non-blank species_comments
-  stopsmbbs <- mbbs %>%
-    filter(species_comments != "") 
-  
-  #fix unicode errors on enter signs, and known typos within the dataset
-  stopsmbbs$species_comments <- stopsmbbs$species_comments %>%
+#' Fix species comments
+#' 
+#' Fix unicode errors on enter signs,
+#' and known typos within the dataset
+#' @param x a character vector of species comments
+#' @importFrom stringr str_replace_all
+fix_species_comments <- \(x) {
```

Comment:
@IJBG - do you think we'll be updating this function on a regular basis (i.e. annually) as new comments are possibly problematic? If so, we may want to separate the logic of the function (string replacement) from the details of what gets replaced. We could have a yaml or csv configuration file that contains the replace/with pairs, which would be read by this function. This would make it so one would update the configuration file with new replacements instead of the function.

We already have a somewhat similar workflow with [ `excluded_submissions.yaml`](https://github.com/nc-minibbs/mbbs/blob/master/inst/excluded_submissions.yml).

Reply:
No, the set of ebird imports that have relevant species comments are stable since it's only pre-2019.

Reply:
Ok. We can leave it as is.